docs/synapse/devguide.rst

```diff
@@ -0,0 +1,331 @@
+
+Remote Cortex Access
+====================
+
+A Cortex, like most synapse components, provides two mechanisms for remote API calls.  The HTTP/REST API and the Telepath API.  For additional documentation on the Cortex HTTP API, see :ref:`http-api`.  This guide will cover remote API calls using Telepath.
+
+Telepath is an asynchronous, high-performance, streaming oriented, RPC protocol.  It is designed for minimum development effort and maximum performance.  Data is serialized using the highly efficient Message_Pack_ format which is not only more size efficient than JSON, but allows serialization of binary data and supports incremental decoding for use in stream based protocols.
+
+Telepath allows a client to connect to a Python object shared on a remote server and, in most instances, call methods as though the object were local.  However, this means all arguments and return values must be serializable using Message Pack.
+
+To connect to a remote object, the caller specifies a URI to connect and construct a Telepath Proxy.  In the following examples, we will assume a Cortex was previously setup and configured with the user ``visi`` and the password ``secretsauce`` running on port ``27492`` on the host ``1.2.3.4``.
+
+Making a simple call
+--------------------
+
+Once a Telepath proxy is connected, most methods may simply be called as though the object were local.  For example, the ``getModelDict`` method on the ``CoreApi`` returns a Python dictionary containing the details of the data model in the remote Cortex.::
+
+    import asyncio
+    import synapse.telepath as s_telepath
+
+    async def main():
+
+        async with await s_telepath.openurl('tcp://visi:secretsauce@1.2.3.4:27492/') as core:
+
+            model = await core.getModelDict()
+
+            for form in model.get('forms'):
+                dostuff()
+
+    if __name__ == '__main__':
+        asyncio.run(main())
+
+Like many objects in the Synapse ecosystem, a Telepath proxy inherits from ``synapse.lib.base.Base``.  This requires the ``fini`` method to be called to release resources and close sockets.  In the example above, we use the async context manager implemented by the ``Base`` class (``async with``) to ensure that the proxy is correctly shutdown.  However, Telepath is designed for long-lived Proxy objects to minimize API call delay by using existing sockets and sessions.  A typical app will create a telepath proxy during initialization and only create a new one in the event that the remote Telepath server is restarted.
+
+The above example also demonstrates that Telepath is designed for use with Python 3.7 asyncio.  However, it is possible to construct and use a Telepath proxy from non-async code.::
```

Comment:
`However, the Telepath proxy can also be constructed and used transparently from non-async code as seen below.` ???